addons/fleet/models/fleet_vehicle.py

```diff
@@ -200,36 +198,31 @@ def _compute_contract_reminder(self):
         for record in self:
             overdue = False
             due_soon = False
-            total = 0
-            name = ''
             state = ''
-            for element in record.log_contracts:
-                if element.state in ('open', 'expired') and element.expiration_date:
-                    current_date_str = fields.Date.context_today(record)
-                    due_time_str = element.expiration_date
-                    current_date = fields.Date.from_string(current_date_str)
-                    due_time = fields.Date.from_string(due_time_str)
-                    diff_time = (due_time - current_date).days
-                    if diff_time < 0:
-                        overdue = True
-                        total += 1
-                    if diff_time < delay_alert_contract:
-                        due_soon = True
-                        total += 1
-                    if overdue or due_soon:
-                        log_contract = self.env['fleet.vehicle.log.contract'].search([
-                            ('vehicle_id', '=', record.id),
-                            ('state', 'in', ('open', 'expired'))
-                            ], limit=1, order='expiration_date asc')
-                        if log_contract:
-                            # we display only the name of the oldest overdue/due soon contract
-                            name = log_contract.name
-                            state = log_contract.state
+            log_contract = self.env['fleet.vehicle.log.contract'].search(
+                [
+                    ('vehicle_id', '=', record.id),
+                    ('state', 'in', ('futur', 'open', 'expired')),
+                ],
+                limit=1, order='expiration_date desc')
```

Comment:
To limit the queries, the search method should be a read_group outside of the for loop with records grouped by vehicle_id to find the log_contract more easily.

Reply:
yeah I just remade the method from scratch because it was really really not optimized at all